lib/src/test/java/com/dynatrace/metric/util/DynatraceMetadataEnricherTest.java

```diff
@@ -19,50 +19,64 @@
 import java.io.FileReader;
 import java.io.IOException;
 import java.io.StringReader;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
 import org.junit.jupiter.api.Test;
 import org.mockito.MockedStatic;
 import org.mockito.Mockito;
 
-public class DynatraceMetadataEnricherTest {
+class DynatraceMetadataEnricherTest {
 
   @Test
-  public void validMetrics() {
+  void validProperties() {
+    Properties properties = new Properties();
+    properties.setProperty("prop.a", "value.a");
+    properties.setProperty("prop.b", "value.b");
+
     ArrayList<Dimension> entries =
-        new ArrayList<>(
-            DynatraceMetadataEnricher.parseDynatraceMetadata(
-                Arrays.asList("prop.a=value.a", "prop.b=value.b")));
-
-    assertEquals("prop.a", entries.get(0).getKey());
-    assertEquals("value.a", entries.get(0).getValue());
-    assertEquals("prop.b", entries.get(1).getKey());
-    assertEquals("value.b", entries.get(1).getValue());
+        new ArrayList<>(DynatraceMetadataEnricher.createDimensionList(properties));
+
+    // Has one entry with key "prop.a"
+    assertEquals(1, entries.stream().filter(entry -> entry.getKey().equals("prop.a")).count());
+
+    // Entry with key "prop.a" has value "value.a"
+    assertEquals(
+        "value.a",
+        entries.stream()
+            .filter(entry -> entry.getKey().equals("prop.a"))
+            .findFirst()
+            .get()
+            .getValue());
+
+    // Has one entry with key "prop.p"
+    assertEquals(1, entries.stream().filter(entry -> entry.getKey().equals("prop.b")).count());
+
+    // Entry with key "prop.b" has value "value.b"
+    assertEquals(
+        "value.b",
+        entries.stream()
+            .filter(entry -> entry.getKey().equals("prop.b"))
+            .findFirst()
+            .get()
+            .getValue());
   }
 
   @Test
-  public void invalidMetrics() {
-    assertTrue(
-        DynatraceMetadataEnricher.parseDynatraceMetadata(Collections.singletonList("key_no_value="))
-            .isEmpty());
-    assertTrue(
-        DynatraceMetadataEnricher.parseDynatraceMetadata(Collections.singletonList("=value_no_key"))
-            .isEmpty());
-    assertTrue(
-        DynatraceMetadataEnricher.parseDynatraceMetadata(
-                Collections.singletonList("==============="))
-            .isEmpty());
-    assertTrue(
-        DynatraceMetadataEnricher.parseDynatraceMetadata(Collections.singletonList("")).isEmpty());
-    assertTrue(
-        DynatraceMetadataEnricher.parseDynatraceMetadata(Collections.singletonList("=")).isEmpty());
-    assertTrue(DynatraceMetadataEnricher.parseDynatraceMetadata(Collections.emptyList()).isEmpty());
+  void invalidProperties() {
```

Comment:
You could parameterize the test with the values below. This way when you run the tests, their names show in a nice list :)